kernel/kernel-api/src/main/java/io/delta/kernel/internal/snapshot/SnapshotManager.java

```diff
@@ -364,12 +419,32 @@ private SnapshotImpl getSnapshotAtInit(Engine engine)
                     "Will search for the checkpoint files directly.", tablePath);
         }
         Optional<LogSegment> logSegmentOpt =
-            getLogSegmentFrom(engine, lastCheckpointOpt);
+                getLogSegmentFrom(engine, lastCheckpointOpt);
 
         return logSegmentOpt
-            .map(logSegment -> createSnapshot(
-                logSegment, engine))
-            .orElseThrow(() -> new TableNotFoundException(tablePath.toString()));
+                .map(logSegment -> getSnapshot(engine, logSegment))
+                .orElseThrow(() -> new TableNotFoundException(tablePath.toString()));
+    }
+
+    private SnapshotImpl getSnapshot(Engine engine, LogSegment initialSegmentForNewSnapshot) {
+        SnapshotImpl newSnapshot = createSnapshot(initialSegmentForNewSnapshot, engine);
+
+        Optional<CommitCoordinatorClientHandler> newCommitCoordinatorClientHandlerOpt =
+                newSnapshot.getCommitCoordinatorClientHandlerOpt(engine);
+        boolean shouldUpdate = newCommitCoordinatorClientHandlerOpt.isPresent();
```

Comment:
s/shouldUpdate/shouldRegenerateWithCoordinatedCommits ? If not, you can just move the .isPresent() to the if () and remove the additional boolean variable.